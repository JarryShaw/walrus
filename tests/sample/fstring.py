print(f'{(x := "awesome")!r:20}', x)


class A:
    locals = 555
    print(f'{(y := 666)}')
    print(y)
    print(f'''L1 -> {f"""L2 -> {f'L3 -> {f"L4 -> {(z := chr(77))!r} <- L4"} <- L3'} <- L2"""} <- L1''')
    print(z)
    print(f'see -> {"header " + f"{(w := chr(88))}" + " footer"} <- see')
    print(w)
    print(locals)


print(globals().get('y'))
print(globals().get('z'))
print(globals().get('w'))
print(callable(locals))
