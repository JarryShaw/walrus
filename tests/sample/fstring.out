'awesome'            awesome
666
666
L1 -> L2 -> L3 -> L4 -> 'M' <- L4 <- L3 <- L2 <- L1
M
555
None
None
True
